### ISSUE TITLE
Support task sorting in task list

### ISSUE THREAD

**Issue by rombert** [devsprint/org.eclipse.mylyn.github#5]
Github query results can be sorted, but they come back more or less random when querying the repository.

It would be nice to support some form of sorting queries.

**Comment by devsprint** [comment]
I have implemented ordering of the collections that are passed to collector in performQuery() method. But something is happening after TaskData is created and submitted through collector. The ordering is not reflected in Task List...
https://github.com/devsprint/org.eclipse.mylyn.github/commit/623decc7d4fba2f531113f6b88099924758bade2

**Comment by rombert** [comment]
I _think_ that what is happening is that the task list sorting kicks in . What you probably need to do is to map the github attribute to task data attributes so that tasks can participate in task list sorting.

See Task List-> Sort for how tasks can be sorted.
See TaskAttributeMapper.mapToRepositoryKey to see how the mapping should be implemented.

**Comment by devsprint** [comment]
I think now I understand to what you are referring with sorting.... I went into a different direction. I haven't seen the Sort... option in my Task List View.... basically it doesn't matter how I'm sorting the collection when I'm gathering the issues from repository and there is no point the add the sort option in query dialog. It's important to make sure that I'm properly matching GitHub attribute with Mylyn known attributes ... but still how do I support GitHub specific attributes like labels and Votes....?

**Comment by devsprint** [comment]
so  the answer is in  your comment  https://github.com/devsprint/org.eclipse.mylyn.github/issues/#issue/5/comment/845064 comment....
now I have to remove most of the work that I did on Saturday.... ;)

**Comment by rombert** [comment]
Well, my request was pretty vague, sorry. I should've mentioned this before.

**Comment by devsprint** [comment]
nope, your request was not vague, I have didn't quite paid enough attention. I was closed to re-invent the wheel.....
